mitodl/mit-open
# Adds base infra for the Unified Ecommerce frontend

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/5611

### Description (What does it do?)

We're going to develop the frontend for Unified Ecommerce in Learn, and this adds some necessary support stuff to that:
- Adds an `EcommercePages` folder so we can store the page components for the UE frontend
- Adds an `EcommerceFeature` component so we can easily feature flag out the UE stuff (especially now, since we'll be doing things to begin with that don't have design yet)
- Adds a `CartPage` placeholder page, just so there's _something_ to go to
- Adds routing for `/cart/` to go to the UE code - this requires a logged-in session and an enabled feature flag to access

### How can this be tested?

Automated tests should pass.

To manually test, you'll need to make sure your instance is hitting PostHog - so it needs a valid set of keys. You can either hit the "MIT Learn Test" project that's in the OL PostHog account, or you can spin up your own PostHog account, or you can spin up your own instance of PostHog locally; any of these will work. Then, you'll need to go into Feature Flags and create a flag that just releases a boolean value called `enable-ecommerce`. 

Test by loading `<site root>/cart/`:

|    | Flag On | Flag Off |
|---|---|---|
| Authenticated | Placeholder page shown | ForbiddenError |
| Not Authenticated | Redirect to login | Redirect to login |

### Additional Context

This doesn't add the API client for UE - we'll do that in a separate PR because it'll probably require some tweaking in the UE code. So, we're just focusing on getting UE some space set up to move into.
